lib/Markdown/Perl.pm

```diff
@@ -324,6 +324,8 @@ optionally a C<title> key containing the title of the key. The hash-ref can also
 contain a C<content> key, in which case its value should be a span of HTML which
 will replace whatever would have been used for the link content.
 
+C<yaml_metadata>: this hook will trigger if there is valid (!) YAML metadata in the file and will give you a YAML::Tiny object as an argument.
```

Comment:
Can you also write that the hook can abort the processing of the document by dying and that the error will be returned as-is from the `convert` call (and test that this is true)?

I see that you have another version where you abort with a false value. I think that dying is better because there is less risk to returns a false value by mistake and, anyway, the convert call must also let the user know that it didn’t process the document so I guess that dying is the simpler option.

Reply:
I can implement the die() but it feels a bit weird as a user of the module to die() if i don't want to process the markdown further. It's not an irrecoverable error in the hook per se, it's wanted behaviour (at least for me) so return true or false as a ways to convey "process further or stop" sounds more logical to me. 

But either way will work. If you prefer a die() to happen from the hook, I'll implement it as such and update the docs with the approach taken.

Reply:
Imagine a hook like `sub { $global_value = $_[0][0]{property} }`, this would work until the property is not defined in the document and now the document would silently not be processed.

I understand that from the point of view of the hook controlling the processing using the return value makes sense, but from the point of view of the `convert` function just silently returning nothing is dangerous and difficult to debug. So `convert` should probably die if the processing is aborted (which is a common Perl approach). If `convert` dies, the simplest is to just propagate an error created by the hook, that way the user can decide what is the exception. You could even imagine something where the hook is `sub { die $_[0] }` so that you can catch the YAML object from outside of `convert` if you only need that.

Don’t see this as the hook dying from an error, just see this as the hook communicating to the module that the processing should be aborted.